src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs

```diff
@@ -115,7 +115,7 @@ public ServiceAccountRepository(PortalDbContext portalDbContext)
             .Where(serviceAccount =>
                 serviceAccount.Id == serviceAccountId &&
                 serviceAccount.Identity!.UserStatusId == UserStatusId.ACTIVE &&
-                serviceAccount.Identity.CompanyId == companyId)
+                serviceAccount.Identity.CompanyId == companyId && (serviceAccount.CompaniesLinkedServiceAccount!.Owners == companyId || serviceAccount.CompaniesLinkedServiceAccount!.Provider == companyId))
```

Comment:
The check of the identity.CompanyId can be removed since we do the check for a matching company with the companiesLinkedServiceAccount

```suggestion
                serviceAccount.CompaniesLinkedServiceAccount!.Owners == companyId || serviceAccount.CompaniesLinkedServiceAccount!.Provider == companyId)
```

Reply:
fixed